pkg/rtc/room.go

```diff
@@ -575,14 +575,15 @@ func (r *Room) RemoveParticipant(identity livekit.ParticipantIdentity, pID livek
 		return
 	}
 
+	r.removeParticipantWorkerLocked(p)
```

Comment:
we can't skip cleaning this up if the new participant is added before the old one is removed or we'll leak the worker

Reply:
good catch, missed that.